src/constants.h

```diff
@@ -8,6 +8,8 @@
 #ifndef CONSTANTS_H_
 #define CONSTANTS_H_
 
+/** Threshold to determine if a float equals to 0.0 */
+#define FLT_EPSILON 1.0E-50
```

Comment:
Did you do some testing ? How about using std::numeric_limits<T>::epsilon() ?

Reply:
No, I didn't do much testing, but I found that std::numeric_limits<float>::epsilon() = 1.19209e-07 and std::numeric_limits<double>::epsilon() = 2.22045e-16, while we already have some very small 1e-20 values in the code. So I chose a smaller value 1E-50.